src/store.rs

```diff
@@ -14,25 +14,17 @@ pub trait TextureStore {
 	/// Add a texture to the store
 	fn add(&mut self, asset: Image) -> Handle<Image>;
 	/// Get a texture from the store
-	fn get<H: Into<HandleId>>(&self, handle: H) -> Option<&Image>;
+	fn get<H: Into<Handle<Image>>>(&self, handle: H) -> Option<&Image>;
```

Comment:
Note that this is `Handle<Image>` now... I wasn't sure whether we would prefer to store `AssetId<Image>` instead